xcdat/regridder/xesmf.py

```diff
@@ -0,0 +1,192 @@
+import xarray as xr
+import xesmf as xe
+
+from xcdat.logger import setup_custom_logger
+from xcdat.regridder.base import BaseRegridder
+
+logger = setup_custom_logger(__name__)
+
+VALID_METHODS = [
+    "bilinear",
+    "conservative",
+    "conservative_normed",
+    "patch",
+    "nearest_s2d",
+    "nearest_d2s",
+]
+
+VALID_EXTRAP_METHODS = ["inverse_dist", "nearest_s2d"]
+
+
+class XESMFRegridder(BaseRegridder):
+    def __init__(
+        self,
+        input_grid: xr.Dataset,
+        output_grid: xr.Dataset,
+        method: str,
+        periodic: bool = False,
+        extrap_method: str = None,
+        extrap_dist_exponent: float = None,
+        extrap_num_src_pnts: int = None,
+        **options,
+    ):
+        """
+        Wrapper class for xESMF regridder class.
+
+        Available options: ``method``, ``periodic``, ``extrap_method``, ``extrap_dist_exponent``, ``extrap_num_src_pnts``
+
+        Parameters
+        ----------
+        input_grid : xr.Dataset
+            Contains source grid coordinates.
+        output_grid : xr.Dataset
+            Contains desintation grid coordinates.
+        method : str
+            Regridding method. Options are
+               - bilinear
+               - conservative
+               - conservative_normed
+               - patch
+               - nearest_s2d
+               - nearest_d2s
+        periodic : bool
+            Treat longitude as periodic. Used for global grids.
+        extrap_method : str
+            Extrapolation method. Options are
+               - inverse_dist
+               - nearest_s2d
+        extrap_dist_exponent : float
+            The exponent to raise the distance to when calculating weights for the extrapolation method.
+        extrap_num_src_pnts : int
+            The number of source points to use for the extrapolation methods that use more than one source point.
```

Comment:
```suggestion
        extrap_dist_exponent : float
            The exponent to raise the distance to when calculating weights for 
            the extrapolation method.
        extrap_num_src_pnts : int
            The number of source points to use for the extrapolation methods 
            that use more than one source point.
```

Fit docstring within 80 character line ruler